presto-orc/src/main/java/com/facebook/presto/orc/reader/IntArrayBlockSupplier.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc.reader;
+
+import com.facebook.presto.orc.OrcCorruptionException;
+import com.facebook.presto.orc.OrcDataSourceId;
+import com.facebook.presto.orc.stream.BooleanInputStream;
+import com.facebook.presto.orc.stream.LongInputStream;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.IntArrayBlock;
+import com.facebook.presto.spi.block.RunLengthEncodedBlock;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import static com.google.common.base.Verify.verify;
+import static java.lang.Math.toIntExact;
+
+public class IntArrayBlockSupplier
+        implements BlockSupplier
+{
+    @Override
+    public Block provideFromDirectStream(OrcDataSourceId dataSourceId, int batchSize, BooleanInputStream presentStream, LongInputStream dataStream)
+            throws IOException
+    {
+        if (presentStream == null) {
+            if (dataStream == null) {
+                throw new OrcCorruptionException(dataSourceId, "Value is not null but data stream is not present");
+            }
+            return populateNonNullBlock(batchSize, dataStream);
+        }
+        else {
```

Comment:
We generally avoid having a "redundant else" (where the "if" block returns or throws unconditionally). IntelliJ has an inspection for this (under *Java | Control flow issues*).

Reply:
Have updated the code to the required format